synapse/lib/stormtypes.py

```diff
@@ -9206,13 +9206,11 @@ def _ts_to_reqdict(ts):
                 'creator': self.runt.user.iden
                 }
 
-        iden = kwargs.get('iden')
-        if iden:
-            cdef['iden'] = iden
+        for k in ['iden', 'name', 'doc']:
+            if len(kwargs.get(k) or '') > 0:
+                cdef[k] = kwargs[k]
 
-        view = kwargs.get('view')
-        if not view:
-            view = self.runt.view.iden
+        view = kwargs.get('view', self.runt.view.iden)
```

Comment:
if `view` could be passed in as a storm `View` object, we'd need to call `toprim` on the kwarg value. We should make sure that any change here is also made in `_methCronAdd`.

Reply:
Looking over `toprim` for the first time and I'm curious:  is there any reason we don't run all argument values through it for normalization after they're passed in?

Reply:
Thank for pointing that out.  I completely whiffed on seeing `self.runt.view.iden` ..not sure how I got down the `--view $lib.view.get().iden` track but that was totally...off-track.